workspace/welcome-to-xod/111-internet/patch.xodp

```diff
@@ -0,0 +1,239 @@
+{
+  "comments": [
+    {
+      "content": "Now you got a remote analog gauge for the data that you send to the cloud. You can place it on your desktop and watch for the moisture on your greenhouse in the countryside.\n\nAlso, another device might turn on or off the water pump, depending on the same data. Moreover, you can send a status of a water pump to the cloud and add some nodes here to indicate that the water pump is working.",
+      "id": "SyoB37C3I",
+      "position": {
+        "units": "slots",
+        "x": 11,
+        "y": 5
+      },
+      "size": {
+        "height": 2,
+        "units": "slots",
+        "width": 8
+      }
+    },
+    {
+      "content": "# Internet On Hardware\n\nYou might be familiar with the `tethering-inet` node and [XOD Cloud](https://xod.io/docs/guide/getting-started-with-feeds/?utm_source=ide&utm_medium=ide_comment&utm_campaign=tutorial) from previous chapters of the tutorial. In this chapter, we make an analog gauge for the data that you've sent before.\n\nThe `xod-cloud/feeds` standard library contains nodes to retrieve the last value from your feed with different types: `cloud-boolean`, `cloud-string`, and `cloud-number`. Since we're sending number values, we are using the latest one from the listed nodes.\n\nOn the left side, there are nodes to react on the data by rotating servo on the right side — nodes to update the value and check out how your servo will respond.\n\nNote that in the tutorial, we named a feed \"moisture\". Usually, the moisture is measured in percentages. So we expect values in a range from 0.00 to 1.00. If you use another range, place the `map` node before the `VAL` input of a `servo` node.\n\n## Excercise\n\n1. Bind a proper `PORT` to the `servo` node\n2. Upload the program with turned on debug mode.\n3. Update `tweak-number` value and check out how your servo responds.\n4. Place a `map` node between `cloud-number` and `servo` if needed.",
```

Comment:
Grammar. Content: we definitely not making a device for the data published previously.

```suggestion
      "content": "# Internet-driven Hardware\n\nIn this chapter, we’ll make an analog gauge for the data that is pushed to the cloud in the same way you already saw in one of the previous chapters.\n\nThe `xod-cloud/feeds` standard library provides nodes to retrieve the last value from your feed with different types: `cloud-boolean`, `cloud-string`, and `cloud-number`. Since we're sending number values, we pick `cloud-number`.\n\nOn the left side, nodes react to the data changes by rotating a servo. On the right side, you see nodes to update the value and check out how your servo responds.\n\nNote that on both sides, we named the feed as \"moisture\", they should match. Usually, the moisture is measured as a percentage. So we expect values in a range from 0.00 to 1.00. If you use another range, place the `map` node before the `VAL` input of a `servo` node.\n\n## Excercise\n\n1. Bind a proper `PORT` to the `servo` node\n2. Upload the program with the debug mode turned on.\n3. Update `tweak-number` value and check out how your servo responds.\n4. Place a `map` node between `cloud-number` and `servo` if needed.",
```